packages/flutter/lib/src/painting/box_decoration.dart

```diff
@@ -123,7 +123,7 @@ class BoxDecoration extends Decoration {
   ///
   /// Applies only to boxes with rectangular shapes; ignored if [shape] is not
   /// [BoxShape.rectangle].
-  final BorderRadius borderRadius;
+  final BorderRadiusGeometry borderRadius;
```

Comment:
Sure you don't want to also rename borderRadius to borderRadiusGeometry?

Reply:
Yes. That would be a breaking change, for one. But also it's inconsistent with how we do this kind of thing elsewhere. The fact that the "FooGeometry" classes exist is mostly a detail most users will never care about.